muse_payment_adaptor/api_interface/muse_api_client.py

```diff
@@ -0,0 +1,158 @@
+        """
+        try:
+            request = self._build_request(bill, payment_desc)
+            request_model = self._store_request(request)
+            # TODO update model after submit?
+            return request_model, self._submit(request)
+        except BaseException as e:
+            logger.error("Error while sending MUSE request", exc_info=e)
+            return None, None
+
+    def _build_request_message(self, bill: Bill, description: str) -> BulkyPaymentSubmissionRequestMessage:
+        message_header = self._build_request_message_header(bill)
+        message_content = self._build_request_message_summary(bill, description)
+        pay_list = self._build_request_pay_list(bill)
+        return BulkyPaymentSubmissionRequestMessage(
+            messageHeader=message_header,
+            paymentSummary=message_content,
+            payList=pay_list,
+        )
+
+    def _build_request(self, bill: Bill, description: str) -> BulkyPaymentSubmissionRequest:
+        message = self._build_request_message(bill, description)
+        signature = self._build_message_signature(message)
+        return BulkyPaymentSubmissionRequest(
+            message=message,
+            digitalSignature=signature
+        )
+
+    def _build_request_message_header(self, bill: Bill) -> BulkyPaymentMessageHeader:
+        return BulkyPaymentMessageHeader(
+            sender=MusePaymentAdaptorConfig.muse_api_sender,
+            receiver=MusePaymentAdaptorConfig.muse_api_receiver,
+            msgId=str(bill.uuid),
+            messageType=MusePaymentAdaptorConfig.muse_bp_header_message_type,
+            paymentType=MusePaymentAdaptorConfig.muse_bp_header_payment_type,
+        )
+
+    def _build_request_message_summary(self, bill: Bill, description: str) -> BulkyPaymentSummary:
+        return BulkyPaymentSummary(
+            institutioncode=MusePaymentAdaptorConfig.muse_bp_message_institution_code,
+            totalAmount=bill.amount_net,
+            referenceNo=str(bill.uuid),
+            paymentDesc=description,
+            applyDate=str(datetime.now()),
+            noofTransaction=bill.line_items_bill.count(),
+            isSTP=False,
+        )
+
+    def _build_request_pay_list(self, bill: Bill) -> List[BulkyPaymentPayListElement]:
+        pay_list = []
+        for bill_item in bill.line_items_bill.all():
+            pay_list.append(self._build_request_pay_list_element(bill_item))
+        return pay_list
+
+    def _build_request_pay_list_element(self, line_item: BillItem) -> BulkyPaymentPayListElement:
+        return BulkyPaymentPayListElement(
+            paymentChannel='',
+            payeeCode='',
+            payeeName='',
+            payeeAccountNumber='',
+            payeeAccountName='',
+            payeeBankName='',
+            payeeBankBic='',
+            amount=line_item.amount_net,
+        )  # TODO API call to request this info ?
```

Comment:
Any action needed?

Reply:
We have to define how we get the payee info for EOs. This is not in scope.